hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -687,14 +685,16 @@ protected void removeAllInternal(Predicate predicate) {
                 OperationFactory operation = operationProvider
                         .createPartitionWideEntryWithPredicateOperationFactory(name, ENTRY_REMOVING_PROCESSOR,
                                 partitionPredicate.getTarget());
-                Data partitionKey = toDataWithStrategy(partitionPredicate.getPartitionKey());
-                int partitionId = partitionService.getPartitionId(partitionKey);
 
                 // invokeOnPartitions is used intentionally here, instead of invokeOnPartition, since
                 // the later one doesn't support PartitionAwareOperationFactory, which we need to use
                 // to speed up the removal operation using global indexes
                 // (see PartitionWideEntryWithPredicateOperationFactory.createFactoryOnRunner).
-                operationService.invokeOnPartitions(SERVICE_NAME, operation, singletonList(partitionId));
+                operationService.invokeOnPartitions(
+                        SERVICE_NAME,
+                        operation,
+                        partitionService.getPartitionIdSet(partitionPredicate.getPartitionKeys())
```

Comment:
In the old implementation, we were using the per map config partitioning strategy with `toDataWithStrategy`, but now the implementation uses regular `toData` with a global partitioning strategy. I feel like this can be a bit problematic. WDYT @ahmetmircik ?

Reply:
`partitionKey` was only ever used to get the `partitionId` to execute on.

From what I can see the data object isn't used for anything else.

I might be missing something, but because we now have the `getPartitionKeys()` method available this is no longer needed.

Reply:
`toDataWithStrategy` is used to find correct partition-ids. 
The returned object keys from `getPartitionKeys` method must be converted to data by using `toDataWithStrategy`, then partition-id-set must be populated by using these data.